src/controller/python/chip/interaction_model/Delegate.cpp

```diff
@@ -79,6 +80,39 @@ CHIP_ERROR PythonInteractionModelDelegate::CommandResponseProcessed(const app::C
     return CHIP_NO_ERROR;
 }
 
+void PythonInteractionModelDelegate::OnReportData(const app::ReadClient * apReadClient, const app::ClusterInfo & aPath,
+                                                  TLV::TLVReader * apData, Protocols::InteractionModel::ProtocolCode status)
+{
+    if (onReportDataFunct != nullptr)
+    {
+        CHIP_ERROR err = CHIP_NO_ERROR;
+        TLV::TLVWriter writer;
+        uint8_t writerBuffer[CHIP_CONFIG_DEFAULT_UDP_MTU_SIZE];
+        writer.Init(writerBuffer, sizeof(writerBuffer));
+        // When the apData is nullptr, means we did not receive a valid attribute data from server, status will be some error
+        // status.
+        if (apData != nullptr)
+        {
+            TLV::TLVReader tmpReader;
+            tmpReader.Init(*apData);
+            // The Copy operation should success since:
+            // - We used a buffer that is large enough
+            // - The writer is in a clean state.
+            err = writer.CopyElement(TLV::AnonymousTag, tmpReader);
+        }
+        // We failed to dump the TLV data to buffer, so we cannot pass valid data to the Python side, this should be a internal
```

Comment:
comment seems out of place - place it on the else branch and add a log in the else branch otherwise we silently drop the error.